.github/workflows/build_etiss.yml

```diff
@@ -0,0 +1,133 @@
+name: Build ETISS
+
+on:
+  repository_dispatch:
+    types: [s4e-cdl-event, seal5-event]
+#push:
+ #   branches: [ "master" ]
+ # pull_request:
+ #   branches: [ "master" ]
+
+env:
+  BUILD_TYPE: Release
+  SEAL5_PREBUILT_DIR: /home/runner/work/etiss/etiss/seal5_prebuilt/release
+  RISCV_GCC_NAME: riscv32-unknown-elf
+  ARCH: rv32imc_xs4emac
+  ETISS_ARCH: RV32IMCXS4EMAC
+  PROG: s4emac
+  ABI: ilp32
+
+jobs:
+  setup-run-M2ISAR:
+    runs-on: ubuntu-22.04
+    
+    steps:
+    - name: Install dependencies
+      run: |
+        sudo apt-get update
+        pip install virtualenv
+    
+    - name: Setup M2-ISA-R
+      run: |
+       git config --global url."https://github.com/".insteadOf git@github.com:
+       git config --global url."https://".insteadOf git://
+       git clone https://github.com/PhilippvK/riscv-coredsl-extensions.git --branch etiss --recursive
+       git clone https://github.com/tum-ei-eda/etiss_arch_riscv.git --recursive --branch coredsl_exceptions
+       git clone https://github.com/tum-ei-eda/M2-ISA-R.git --branch coredsl2
+       cd M2-ISA-R
+       virtualenv -p python3.10 venv  #Alternative (requires `apt install python3-venv`): python3 -m venv venv
+       source venv/bin/activate
+       pip install -e .
+      
+    - name: Run M2-ISA-R
+      run: |
+        cd M2-ISA-R
+        source venv/bin/activate
+        python -m m2isar.frontends.coredsl2.parser ../riscv-coredsl-extensions/etiss-s4e-mac.core_desc
+        python -m m2isar.backends.etiss.writer ../riscv-coredsl-extensions/gen_model/etiss-s4e-mac.m2isarmodel --separate --static-scalars
+       
+    - name: Upload M2ISAR Model artifacts
+      uses: actions/upload-artifact@v4
+      with:
+         name: m2isar-model-files
+         path: |
+           riscv-coredsl-extensions/gen_output
+            
+  patch-etiss-arch:
+    runs-on: ubuntu-22.04
+    needs: [setup-run-M2ISAR]
+    steps:   
+    - name: Download Patched Etiss
+      uses: actions/download-artifact@v4
+      with:
+        name: m2isar-model-files
+        github-token: ${{ secrets.REPO_ACCESS_TOKEN }}
+       
+   # - uses: actions/checkout@v4
+   #   with:
+   #     path: etiss_source
+   #     ref: coverage
+        
+    - run: |
+       git config --global url."https://github.com/".insteadOf git@github.com:
+       git config --global url."https://".insteadOf git://
+       git clone git@github.com:wysiwyng/etiss.git etiss_source --branch coverage
```

Comment:
This should ideally just clone the etiss repository using the `actions/checkout` action.

To get rid of this, @wysiwyng would need to merge the coverage related commits to upstream etiss.